### PR TITLE
MMG and other MeshingApplication Doxygen improve/Clean up

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ping

**Comment by pooyan-dadvand** [comment]
I think that you should divide this PR in two. Changes in code, not only in doxygen files

**Comment by loumalouomega** [comment]
I thought that, but the changes are minor tweaks, I can separate it in several PR for several files

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
You are adding global enums to Kratos namespace which is not a good practice...

**Comment by loumalouomega** [comment]
Even with enum class?

File: applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h [hunk 1]
```diff
@@ -126,9 +129,13 @@ class InternalVariablesInterpolationProcess
     ///@{
 
     /**
-     * We execute the search relative to the old and new model part
+     * @brief We execute the search relative to the old and new model part
+     * @details There are mainly two ways to interpolate the internal variables (there are three, but just two are behave correctly)
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h:144]: shouldn't we remove the broken one?

File: applications/MeshingApplication/custom_processes/mmg_process.cpp [hunk 2]
```diff
@@ -216,36 +217,40 @@ void MmgProcess<TDim>::InitializeMeshData()
...
-                #pragma omp atomic
-                num_array_elements[0] += 1;
+                num_tetra += 1;
             } else if ((it_elem->GetGeometry()).GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Prism3D6) { // Prisms
-                #pragma omp atomic
-                num_array_elements[1] += 1;
+                num_prisms += 1;
             } else
-                std::cout << "WARNING: YOUR GEOMETRY CONTAINS HEXAEDRON THAT CAN NOT BE REMESHED" << std::endl;
+                std::cout << "WARNING: YOUR GEOMETRY CONTAINS CERTAIN TYPE THAT CAN NOT BE REMESHED" << std::endl;
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/mmg_process.cpp]: not very descriptive...

File: applications/MeshingApplication/custom_processes/mmg_process.cpp [hunk 3]
```diff
@@ -216,36 +217,40 @@ void MmgProcess<TDim>::InitializeMeshData()
...
         }
         
-        if (((num_array_elements[0] + num_array_elements[1]) < elements_array.size()) && mEchoLevel > 0)
-            std::cout << "Number of Elements: " << elements_array.size() << " Number of Tetrahedron: " << num_array_elements[0] << " Number of Prisms: " << num_array_elements[1] << std::endl;
+        if (((num_tetra + num_tetra) < elements_array.size()) && mEchoLevel > 0)
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/mmg_process.cpp:234]: this looks strange. you sure?

**Review by RiccardoRossi — DISMISSED** [review]
please take a look at my comments. at least one looks like a typo (you are summing twice the same var).

I approvie in case it is the expected behaviour

**Comment by loumalouomega** [comment]
I will take into account these considerations, I will do as soon as I finish what I am doing right now

**Comment by loumalouomega** [comment]
@pooyan-dadvand I need to do some changes in the internal_variables_interpolation_process, do you mind if I merge to avoid conflicts and I incorporate your suggestions in this new PR?

File: applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h (hunk 1 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h:144]: OK

File: applications/MeshingApplication/custom_processes/mmg_process.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/mmg_process.cpp]: Ok

**Comment by loumalouomega** [comment]
Done, sorry for the delay

**Comment by loumalouomega** [comment]
@pooyan-dadvand What is missing?

**Comment by pooyan-dadvand** [comment]
Have you applied the comments of @RiccardoRossi ?

What are LST, CPT, etc? IMO a more descriptive name especially for enums is useful.

**Comment by loumalouomega** [comment]
Yes, I applied the changes suggested by @RiccardoRossi. The methods are
described in the doxygen of the Execute method.

**Comment by pooyan-dadvand** [comment]
LST, CPT, etc are not methods and reading the doxygen for understanding a variable is from 80s.

**Comment by loumalouomega** [comment]
OK, I will append a description on place

**Comment by loumalouomega** [comment]
Done, i found I typo too

**Comment by loumalouomega** [comment]
Besides I use Doxygen because I thought it is the proper way to document

**Comment by jcotela** [comment]
@loumalouomega: you should not need to read the comments to understand what the name of your variable means. The days of 8-character-maixmum variable names are gone.

**Comment by loumalouomega** [comment]
I don't have any problem with large name for variables, in the code you can see as I use `constitutive_law_vector` and similar names to describe variables.

 I am just using an acronym for a method, which is the classic name people uses in the literature. I think this is not necessarily wrong approach. We usually write PK2 not Piola-Kirchoff 2; we usually say formulation UP, not mixed displacement-pressure formulation.

**Comment by pooyan-dadvand** [comment]
Thanks for changing the methods but the enums are still there....

**Comment by loumalouomega** [comment]
True, I changed to enum class, but I forgot to move it inside the class, one sec

**Comment by loumalouomega** [comment]
Done, sorry I did the hard part and not the easy one :P

**Comment by loumalouomega** [comment]
@pooyan-dadvand Do I need to do anything else?

**Comment by pooyan-dadvand** [comment]
I would change the enum lst cpt etc to more descriptive names

**Comment by loumalouomega** [comment]
Done

**Review by pooyan-dadvand — APPROVED** [review]
Do we need TRANSFER in the enum names?

**Comment by loumalouomega** [comment]
It is just a descriptive name

**Pull request merged**